File: services/auth-service/src/main/java/com/otterworks/auth/entity/UserSettings.java
package com.otterworks.auth.entity;

import jakarta.persistence.*;
import java.time.Instant;
import java.util.UUID;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "user_settings")
@Getter
@Setter
@NoArgsConstructor
public class UserSettings {

  @Id
  @Column(name = "user_id")
  private UUID userId;

  @Column(nullable = false)
  private boolean notificationEmail = true;

  @Column(nullable = false)
  private boolean notificationInApp = true;

  @Column(nullable = false)
  private boolean notificationDesktop = false;

  @Column(nullable = false, length = 20)
  private String theme = "system";

  @Column(nullable = false, length = 10)
  private String language = "en";

  @Column(nullable = false)
  private Instant updatedAt;

  @PrePersist
  protected void onCreate() {
    updatedAt = Instant.now();
  }

  @PreUpdate
  protected void onUpdate() {
    updatedAt = Instant.now();
  }
}

File: services/auth-service/src/Data/Entities/UserSettings.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OtterWorks.AuthService.Data.Entities;

[Table("user_settings")]
public class UserSettings
{
    [Key]
    [Column("user_id")]
    public Guid UserId { get; set; }

    [Column("notification_email")]
    public bool NotificationEmail { get; set; } = true;

    [Column("notification_in_app")]
    public bool NotificationInApp { get; set; } = true;

    [Column("notification_desktop")]
    public bool NotificationDesktop { get; set; }

    [Required]
    [MaxLength(10)]
    [Column("theme")]
    public string Theme { get; set; } = "system";

    [Required]
    [MaxLength(10)]
    [Column("language")]
    public string Language { get; set; } = "en";
}
